src/hera/cron_workflow.py

```diff
@@ -65,29 +65,32 @@ def __init__(
         self.service = service
         self.parallelism = parallelism
         self.service_account_name = service_account_name
-        self.labels = labels
-        self.namespace = namespace
+        self.labels = labels or {}
```

Comment:
Doing defaults like this could have unintended consequences, as was the case with the command in a `Task` being impossible to override and always being forced to be `python`. I feel either we use defaults via the `labels ={}` or not at all, unless absolutely required.

Reply:
> either we use defaults via the labels ={} or not at all, unless absolutely required.

I am starting to share this sentiment. Perhaps we do it the same way we set the `service_account_name` rather than set default values? The new SDK objects use a bunch of `setattr` dynamically, so we can also use that for labels if they are set. I can also definitely see the potential for unintended consequences - perhaps K8S might use these empty values somehow (generally, not labels necessarily) and behave unexpectedly. 

What do you think about resorting to the use of conditional `setattr(...)` the same way we do it with the `service_account_name` but for labels, and basically any other field that moved from `None` to something else? I now think it's better to check for nullity and set accordingly, which also supports more consistency in usage of the SDK.

Reply:
I think that's probably a good way to go about it, while it is definitely verbose, I think it is more explicit for the end-user. I am though not against using defaults in signatures in certain situations, for example with the python command.

I am intrigued though why the use of `setattr()` vs `self.template.service_account_name = self.service_account_name`. Although, I see that with the new OpenAPI spec, either one is not validated, and there would be no code editor benefit with either. Prob more a personal issue than anything else, I just find string name variables slightly on the edge of Magic Variables.